miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppStorageTest.kt

```diff
@@ -44,7 +45,37 @@ class MiniAppStorageTest {
         verify(localUrlParser, times(1)).getFileName(TEST_URL_FILE)
     }
 
-    private fun getMockedMiniAppStorage() = MiniAppStorage(mock(), mock(), mock())
+    @Test
+    fun `should get consistent path when get path for mini app version`() {
+        val storage = MiniAppStorage(FileWriter(), File(TEST_BASE_PATH))
+
+        storage.getPathMiniApp(TEST_ID_MINIAPP) shouldStartWith storage.getRootPathMiniApp()
+        storage.getRootPathMiniApp() shouldStartWith storage.getBasePathHostApp()
+        storage.getPathMiniAppVersion(TEST_ID_MINIAPP, TEST_ID_MINIAPP_VERSION) shouldStartWith
+                storage.getPathMiniApp(TEST_ID_MINIAPP)
+        storage.getPathMiniAppVersion(TEST_ID_MINIAPP, TEST_ID_MINIAPP_VERSION) shouldEndWith
+                TEST_ID_MINIAPP_VERSION
+    }
+
+    @Test
+    fun `should delete all file data excluding the latest version package`() {
+        val oldFile1 = tempFolder.newFolder("old_package_id_1")
+        val oldFile2 = tempFolder.newFile()
+        val latestPackage = tempFolder.newFolder(TEST_ID_MINIAPP_VERSION)
+
+        runBlocking {
+            miniAppStorage.removeOutdatedVersionApp(
+                TEST_ID_MINIAPP,
+                TEST_ID_MINIAPP_VERSION,
+                tempFolder.root.path)
+            // Deletion run in parallel so it requires time to be completed.
+            delay(1000)
```

Comment:
I'm not sure about this. @corycaywood What do you think?

Reply:
I agree @ts-nitish-a-jain, this test is not safe

Reply:
Yeah, let's find another way of doing this. Actually, in my opinion we should make `MiniAppStorage#removeOutdatedVersionApp` be synchronous (it's already a suspend function), and move the coroutine launch into `RealMiniApp`. What do. you think?

Reply:
it should be launched inside `MiniAppDownloader` not `RealMiniApp`.